frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -343,9 +348,11 @@ pub mod pallet {
 
                 Self::deposit_event(Event::<T>::NewDappStakingEra(next_era));
 
+                // should we return this or should we return a benchmarked weight?
                 consumed_weight + T::DbWeight::get().reads_writes(5, 3)
+                // return T::WeightInfo::on_initialize_new_era();
```

Comment:
This part is actually the reason I mentioned that you should have a few more days to figure this part out.
Because former benchmarks you've implemented are no longer valid for this scenario since we have the new `rotate_staking_info` part.

I will say that this depends on the approach you want to take.

First would be to redesign benchmarks so you can measure this correctly.

Another approach would be to just ditch the benchmarks and use it as I've added it.
(Btw, if you feel bad about loosing the benchmark code - no need. This was a learning exp).

Reply:
Yes I remember you mentioning this! I'll take a look to conceptualize what the new benchmark might look like and will just keep your code here for simplicity.

Reply:
So once you get a bit into this - let's consider how difficult it is (and how complex the code will be) to do proper benchmarking here.

It is also ok to just re-use what we already have (manual weight).

Also, keep in mind that it is possible to benchmark separate functions.
You don't need to benchmark an extrinsic call.
So it's perfectly fine to just do benchmarks on e.g. `rotate_staking_info`.